internal/server/auth/auth-service.go

```diff
@@ -0,0 +1,186 @@
+package auth
+
+import (
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/ShukinDmitriy/GophKeeper/internal/server/models/responses"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/labstack/echo/v4"
+	"go.uber.org/zap"
+)
+
+const (
+	userTokenCookieName    = "user"
+	accessTokenCookieName  = "access-token"
+	refreshTokenCookieName = "refresh-token"
+)
+
+// GetAccessTokenCookieName get access token name
+func GetAccessTokenCookieName() string {
+	return accessTokenCookieName
+}
+
+// GetSigningMethod get signing method name
+func GetSigningMethod() *jwt.SigningMethodHMAC {
+	return jwt.SigningMethodHS256
+}
+
+type Claims struct {
+	ID uint `json:"id"`
+	jwt.RegisteredClaims
+}
+
+type AuthService struct {
+	authUser AuthUser
+}
+
+func NewAuthService(authUser AuthUser) *AuthService {
+	return &AuthService{authUser: authUser}
+}
+
+func GetJWTSecret() string {
+	jwtSecretKey, exists := os.LookupEnv("JWT_SECRET_KEY")
+
+	if !exists {
+		panic("no JWT_SECRET_KEY in .env")
+	}
+	return jwtSecretKey
+}
+
+func (authService *AuthService) GetAccessTokenCookieName() string {
+	return accessTokenCookieName
+}
+
+func (authService *AuthService) GetRefreshTokenCookieName() string {
+	return refreshTokenCookieName
+}
+
+func (authService *AuthService) BeforeFunc(c echo.Context) {
+	accessTokenCookie, err := c.Cookie(authService.GetAccessTokenCookieName())
+	if err == nil {
+		accessUserID, _ := authService.authUser.getUserIDByCookie(c, accessTokenCookie)
+		if accessUserID != nil {
+			return
+		}
+	}
+
+	refreshTokenCookie, err := c.Cookie(authService.GetRefreshTokenCookieName())
+	if err != nil {
+		return
+	}
+
+	if accessTokenCookie == nil && refreshTokenCookie != nil {
+		userID, err := authService.authUser.getUserIDByCookie(c, refreshTokenCookie)
+		if err != nil {
+			return
+		}
+
+		user := authService.authUser.getUserByID(c, *userID)
+
+		err = authService.GenerateTokensAndSetCookies(c, user)
+		if err != nil {
+			return
+		}
+	}
+}
+
+func (authService *AuthService) GenerateTokensAndSetCookies(c echo.Context, user *responses.UserInfo) error {
+	accessToken, accessTokenString, exp, err := authService.generateAccessToken(user)
+	if err != nil {
+		return err
+	}
+
+	_, refreshTokenString, refreshExp, err := authService.generateRefreshToken(user)
+	if err != nil {
+		return err
+	}
+
+	authService.setTokenCookie(c, accessTokenCookieName, accessTokenString, exp)
+	authService.setTokenCookie(c, refreshTokenCookieName, refreshTokenString, refreshExp)
+	c.Set("user", accessToken)
+	authService.setUserCookie(c, user, exp)
+
+	return nil
+}
+
+func (authService *AuthService) GetUserID(c echo.Context) uint {
+	if c.Get("user") == nil {
+		return 0
+	}
+	u, ok := c.Get("user").(*jwt.Token)
+	if !ok {
+		return 0
+	}
+
+	claims, ok := u.Claims.(*Claims)
+	if !ok {
+		return 0
+	}
+
+	return claims.ID
+}
+
+func (authService *AuthService) JWTErrorChecker(c echo.Context, err error) error {
```

Comment:
сигнатура метода диктуется фреймворком. переименовал в _
```// ErrorHandler defines a function which is executed when all lookups have been done and none of them passed Validator
	// function. ErrorHandler is executed with last missing (ErrExtractionValueMissing) or an invalid key.
	// It may be used to define a custom JWT error.
	//
	// Note: when error handler swallows the error (returns nil) middleware continues handler chain execution towards handler.
	// This is useful in cases when portion of your site/api is publicly accessible and has extra features for authorized users
	// In that case you can use ErrorHandler to set default public JWT token value to request and continue with handler chain.
	ErrorHandler func(c echo.Context, err error) error
```